ludwig/features/vector_feature.py

```diff
@@ -0,0 +1,445 @@
+#! /usr/bin/env python
+# coding=utf-8
+# Copyright (c) 2019 Uber Technologies, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import os
+import logging
+
+import numpy as np
+import tensorflow as tf
+
+from collections import OrderedDict
+from ludwig.constants import *
+from ludwig.features.base_feature import BaseFeature
+from ludwig.features.base_feature import InputFeature
+from ludwig.features.base_feature import OutputFeature
+from ludwig.models.modules.dense_encoders import Dense
+from ludwig.utils.misc import get_from_registry
+from ludwig.utils.misc import set_default_value
+from ludwig.models.modules.measure_modules import error as get_error
+from ludwig.models.modules.measure_modules import \
+    absolute_error as get_absolute_error
+from ludwig.models.modules.measure_modules import \
+    squared_error as get_squared_error
+from ludwig.models.modules.measure_modules import r2 as get_r2
+
+logger = logging.getLogger(__name__)
+
+
+class VectorBaseFeature(BaseFeature):
+    def __init__(self, feature):
+        super().__init__(feature)
+        self.type = VECTOR
+
+    preprocessing_defaults = {
+        'missing_value_strategy': FILL_WITH_CONST,
+        'fill_value': 0.0
+    }
+
+    @staticmethod
+    def get_feature_meta(column, preprocessing_parameters):
+        return {
+            'preprocessing': preprocessing_parameters
+        }
+
+    @staticmethod
+    def read_single_vector(row):
+        return [float(x) for x in row.split()]
+
+    @staticmethod
+    def feature_data(column, metadata):
+        vectors = [VectorBaseFeature.read_single_vector(x) for x in column]
+        for v in vectors:
+            if len(v) != metadata['vector_size']:
+                raise ValueError(
```

Comment:
here if you enumerate vectors you can also report the line of the malformed vector

Reply:
Makes sense. Will do